### ISSUE TITLE
Software citation

### ISSUE THREAD

**Issue by rivaquiroga** [programminghistorian/jekyll#1984]
TL; DR: Our guidelines say nothing about citing the software used in a lesson. It should be a requirement.

---

Currently is at the discretion of the author, so we have lessons that cite some or all the software used, and some lessons that don't cite it at all. Citing software is a way of [recognizing its value](https://f1000research.com/articles/9-1257/v2
), so it might be a good idea to make it mandatory. 

If we all agree that we should consider this issue, there are two actions involved.

- On the short run: adding this requirement to the Author Guidelines (as one more bullet point in the section about [references](https://programminghistorian.org/en/author-guidelines#references))
- On the long run: update published lessons so they are properly citing all the software used

**Comment by amsichani** [comment]
Great point @rivaquiroga ! Here is an interesting article from SSI - https://www.software.ac.uk/how-cite-software . Happy to assist on the 'short run' front and for the ' long run' do have in mind that we are planning to add a _testing_ field to yaml #1938 -- also related to #1633 . I will be able to update on this after my mat leave, May/June.

**Comment by walshbr** [comment]
Oh I meant to respond earlier but got distracted. This is a great idea. My only thought was that we might want some guidelines about what we're including on that citation - that link @amsichani might have some sense of this. But I was just thinking there is a big difference between 

Python

and

Python v 3.7.4
Pandas v 1.0.3
Tested on a Mac 2017 running High Sierra

Etc. So we might just want to set some basic template for what we're looking for.

**Comment by rivaquiroga** [comment]
Thanks for the article, @amsichani! That one + https://f1000research.com/articles/9-1257/v2 can be used as a starting point to create our own guidelines.  
@walshbr, yes, totally agree that we need to set a basic template that authors can easily use.

**Comment by drjwbaker** [comment]
> Great point @rivaquiroga ! Here is an interesting article from SSI - https://www.software.ac.uk/how-cite-software . Happy to assist on the 'short run' front and for the ' long run' do have in mind that we are planning to add a _testing_ field to yaml #1938 -- also related to #1633 . I will be able to update on this after my mat leave, May/June.

I was going to point to the very same article! Great minds etc.

**Comment by amsichani** [comment]
Ok. I ve worked on a draft section on software citation to be included in our basic [References](https://programminghistorian.org/en/author-guidelines#references) , Author Guidelines: 

> 
> In order to cite software, which requires citation as a condition of use within a lesson, please provide as much as possible from the information below, following the general "Notes and Bibliography” system found in the [Chicago Manual of Style, 17th Edition ](https://www.chicagomanualofstyle.org/tools_citationguide/citation-guide-1.html ) for endnotes:
> 
> - author 
> - product name
> - date
> - version
> - URL or DOI 

> eg.MeeseeksMachine, Pandas Development Team, Tom Augspurger, William Ayd , jbrockmendel, Pandas, v 1.0.3 (March 17, 2020) , https://pandas.pydata.org/pandas-docs/version/1.0.3/

@walshbr not sure we need to provide the information on the tested environment - this is usually an info provided by the software package itself.

Do let me know how that sounds.

**Comment by acrymble** [comment]
Is this something that should be added to the copy editing guides?

**Comment by amsichani** [comment]
> Is this something that should be added to the copy editing guides?

I think it should be added at the Author Guidelines , yes!

**Comment by rivaquiroga** [comment]
We have [another edition to the Author Guidelines](https://github.com/programminghistorian/jekyll/issues/1712) in process, so I will wait until that one is ready to add this change to the same PR.
